### PR TITLE
UKR-218 Add Options & Variants

### PR DESCRIPTION
**Pull request by SecondeJK**
**Contains**

- Migrations
- Models
- IDE Helper updates
- Basic Unit tests
- Bootstrapped services for new repositories
- Interface definitions for new repositories

### CODE REVIEW

File: tests/Unit/Repository/InventoryOptionGroupItemsRepositoryTest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+        $this->repository = $this->app->make(InventoryOptionGroupItemContract::class);
+    }
+
+    /**
+     * @test
+     */
+    public function canAddItemsToOptionGroup(): void
+    {
+        $inventoryOptionGroupName = 'Musical Equipment';
```
Comment (jump24-luke) [inline tests/Unit/Repository/InventoryOptionGroupItemsRepositoryTest.php:38]: Nitpicking, but as this is only used once should it be like the other creators and just use a string?

File: app/Repository/InventoryOptionGroupItemRepository.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * @var InventoryOptionGroupItem
+     */
+    protected $model;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
```
Comment (dannewnsjump24) [inline app/Repository/InventoryOptionGroupItemRepository.php:23]: It is worth adding the correct type hint here we have been doing it on other repos

- [ ] Add Typehints

File: app/Repository/InventoryOptionGroupRepository.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+use Psr\Log\LoggerInterface;
+
+class InventoryOptionGroupRepository implements InventoryOptionGroupContract
+{
+    use HelperTrait;
+
+    /**
+     * @var InventoryOptionGroup
+     */
+    protected $model;
```
Comment (dannewnsjump24) [inline app/Repository/InventoryOptionGroupRepository.php:18]: Type hints here as well

File: app/Repository/InventoryOptionGroupItemRepository.php (hunk 2 above)
Comment (SecondeJK) [reply app/Repository/InventoryOptionGroupItemRepository.php:23]: I'm going to do a wider review for property type hints here, as this is how some of the other repos look as well

**Pull request merged**